lib/providers/storage.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:concordium_wallet/entities/accepted_terms_and_conditions.dart';
+import 'package:concordium_wallet/state/network.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+
+/// Service for interacting with [Hive].
+class StorageProvider {
+  final Box<AcceptedTermsAndConditions> _acceptedTermsAndConditionBox;
+
+  const StorageProvider._(this._acceptedTermsAndConditionBox);
+
+  static Future<StorageProvider> init() async {
+    await Hive.initFlutter();
+    _registerAdapters();
+    await _openBoxes();
+
+    return StorageProvider._(await Hive.openBox<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table));
+  }
+
+  /// Register all adapters needed for typed boxes.
+  static void _registerAdapters() {
+    Hive.registerAdapter(AcceptedTermsAndConditionsAdapter());
+  }
+
+  /// Opens all boxes asynchronously.
+  static Future<void> _openBoxes() async {
+    final atcFuture = Hive.openBox<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table);
+    await Future.wait([atcFuture]);
+  }
+
+  /// Reads the currently accepted T&C version.
+  AcceptedTermsAndConditions? getAcceptedTermsAndConditions(NetworkName networkName) {
```

Comment:
The terms and conditions are universal and not per network. You only accept them once for a single installation.

Reply:
Yes and we should probably also think of a better way to work with the valid T&C version than fetching it in a network-specific context.